### ISSUE TITLE
Use points cache when requesting multiple aggregation results in a single query.

### ISSUE THREAD

**Issue by joente** [transceptor-technology/siridb-server#76]
SiriDB allows multiple select requests in a single query.

For example:
```
select min(1h) prefix 'min-', max(1h) prefix 'max-' from 'my-series'
```

We already prepared a `points_map` member for caching the result but at the moment SiriDB still reads the requested points for all select requests.

**Comment by joente** [comment]
Implemented caching as following:
- Only one select request:
  - No points cache is used (or created)
- With more than one select request:
  - Points cache is created and each select request, expect the last one, uses a copy of points. (since the original is destroyed)
  - The last select request uses the cached points instead of a copy since the cache is no longer required.
- In case of a failure the cache will be automatically cleared. (this is always done but when successful we have nothing to clean)
- Logic does not depend on the cache. When memory allocation errors occur in the cache logic, everything keeps working as if we had no cache at all

**Will be implemented in version 2.0.21**

**Comment by joente** [comment]
**Timeit results**

Group `test` has 8 series with a total of 107484 points.

Query:
```
timeit select min(1h) prefix 'min-', max(1h) prefix 'max-', mean(1h) prefix 'mean-' from `test`
```

Running 10 times without points cache: 
  - average 0.187 seconds per query

Running 10 times with points cache: 
  - average 0.068 seconds per query (almost 3x faster)
